src/ui/RunEditor.tsx

```diff
@@ -1510,116 +1510,116 @@ export class RunEditor extends React.Component<Props, State> {
+                splitTimeChanged: true,
             },
         });
     }
 
     private handleSegmentTimeChange(event: any) {
-        const valid = this.props.editor.activeParseAndSetSegmentTime(event.target.value);
-        const editor = {
-            ...this.state.editor,
-        };
-        editor.segments[this.state.rowState.index].segment_time = event.target.value;
         this.setState({
             ...this.state,
-            editor,
             rowState: {
                 ...this.state.rowState,
-                segmentTimeIsValid: valid,
+                segmentTime: event.target.value,
+                segmentTimeChanged: true,
             },
         });
     }
 
     private handleBestSegmentTimeChange(event: any) {
-        const valid = this.props.editor.activeParseAndSetBestSegmentTime(event.target.value);
-        const editor = {
-            ...this.state.editor,
-        };
-        editor.segments[this.state.rowState.index].best_segment_time = event.target.value;
         this.setState({
             ...this.state,
-            editor,
             rowState: {
                 ...this.state.rowState,
-                bestSegmentTimeIsValid: valid,
+                bestSegmentTime: event.target.value,
+                bestSegmentTimeChanged: true,
             },
         });
     }
 
     private handleComparisonTimeChange(event: any, comparisonIndex: number) {
-        const comparisonName = this.state.editor.comparison_names[comparisonIndex];
-        const valid = this.props.editor.activeParseAndSetComparisonTime(comparisonName, event.target.value);
-        const editor = {
-            ...this.state.editor,
-        };
-        editor.segments[this.state.rowState.index].comparison_times[comparisonIndex] = event.target.value;
-        const comparisonIsValid = { ...this.state.rowState.comparisonIsValid };
-        comparisonIsValid[comparisonIndex] = valid;
+        const comparisonTimes = { ...this.state.rowState.comparisonTimes };
+        comparisonTimes[comparisonIndex] = event.target.value;
+
+        const comparisonTimesChanged = { ...this.state.rowState.comparisonTimesChanged };
+        comparisonTimesChanged[comparisonIndex] = true;
+
         this.setState({
             ...this.state,
-            editor,
             rowState: {
                 ...this.state.rowState,
-                comparisonIsValid,
+                comparisonTimes,
+                comparisonTimesChanged,
             },
         });
     }
 
     private handleSplitTimeBlur() {
+        if (this.state.rowState.splitTimeChanged) {
+            this.props.editor.activeParseAndSetSplitTime(this.state.rowState.splitTime);
+        }
+
         this.setState({
             ...this.state,
             editor: this.props.editor.stateAsJson(),
             rowState: {
                 ...this.state.rowState,
-                splitTimeIsValid: true,
-            },
+                splitTimeChanged: false
+            }
```

Comment:
I'd like to keep the trailing commas in general. Just letting you know, it's not something you need to fix.